src/se/vidstige/jadb/JadbDevice.java

```diff
@@ -6,31 +6,27 @@
 
 public class JadbDevice {
 	private final String serial;
-	private final Transport transport;
-	private boolean selected = false;
+	private Transport transport;
+    private final ITransportFactory tFactory;
 
-	JadbDevice(String serial, String type, Transport transport) {
+	JadbDevice(String serial, String type, ITransportFactory tFactory) {
 		this.serial = serial;
-		this.transport = transport;
+        this.tFactory = tFactory;
 	}
 
-    static JadbDevice createAny(Transport transport) { return new JadbDevice(transport); }
+    static JadbDevice createAny(JadbConnection connection) { return new JadbDevice(connection); }
 
-    private JadbDevice(Transport transport)
+    private JadbDevice(ITransportFactory tFactory)
     {
         serial = null;
-        this.transport = transport;
+        this.tFactory = tFactory;
     }
 
-    private void ensureTransportIsSelected() throws IOException, JadbException {
-        if (!selected)
-        {
-            selectTransport();
-            selected = true;
+    private void getTransport() throws IOException, JadbException {
+        if(transport!=null && !transport.isClosed()){
```

Comment:
whitespace before parentesis and before brace as well as around not-equals, like so 

if (transport != null && !transport.isClosed()) {